tests/PPC/TestLifting.cpp

```diff
@@ -0,0 +1,801 @@
+         }},
+        {"xer_so",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.xer_flags.so;
+         }},
+        {"xer_ov",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.xer_flags.ov;
+         }},
+};
+
+std::optional<remill::Instruction>
+GetFlows(std::string_view bytes, uint64_t address, uint64_t vle_val) {
+
+  llvm::LLVMContext context;
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  context.enableOpaquePointers();
+#endif
+  auto arch = remill::Arch::Build(&context, remill::OSName::kOSLinux,
+                                  remill::ArchName::kArchPPC);
+  auto sems = remill::LoadArchSemantics(arch.get());
+
+
+  remill::DecodingContext dec_context;
+  dec_context.UpdateContextReg(std::string("VLEReg"), vle_val);
+  CHECK(dec_context.HasValueForReg("VLEReg"));
+  remill::Instruction insn;
+
+  if (!arch->DecodeInstruction(address, bytes, insn, dec_context)) {
+    return std::nullopt;
+  } else {
+    return insn;
+  }
+}
+}  // namespace
+
+using test_runner::TestOutputSpec;
+
+template <typename State, typename = typename std::enable_if_t<
+                              std::is_base_of<ArchState, State>::value>>
+class TestSpecRunner {
+ private:
+  test_runner::LiftingTester lifter;
+  uint64_t tst_ctr;
+  test_runner::random_bytes_engine rbe;
+  llvm::support::endianness endian;
+
+ public:
+  TestSpecRunner(llvm::LLVMContext &context, remill::ArchName archname)
+      : lifter(test_runner::LiftingTester(context, remill::OSName::kOSLinux,
+                                          archname)),
+        tst_ctr(0),
+        endian(lifter.GetArch()->MemoryAccessIsLittleEndian()
+                   ? llvm::support::endianness::little
+                   : llvm::support::endianness::big) {}
+
+  void RunTestSpec(const TestOutputSpec<State> &test) {
+    std::stringstream ss;
+    ss << "test_disas_func_" << this->tst_ctr++;
+
+    auto maybe_func =
+        lifter.LiftInstructionFunction(ss.str(), test.target_bytes, test.addr);
+
+
+    CHECK(maybe_func.has_value());
+    auto lifted_func = maybe_func->first;
+
+    auto new_mod = llvm::CloneModule(*lifted_func->getParent());
+    remill::OptimizeBareModule(new_mod.get());
+
+    auto justFuncMod =
+        std::make_unique<llvm::Module>("", new_mod->getContext());
+
+    auto new_func = test_runner::CopyFunctionIntoNewModule(
+        justFuncMod.get(), lifted_func, new_mod);
+    State st = {};
+
+
+    test.CheckLiftedInstruction(maybe_func->second);
+    test_runner::RandomizeState(st, this->rbe);
+
+    test.SetupTestPreconditions(st);
+    auto mem_hand = std::make_unique<test_runner::MemoryHandler>(this->endian);
+
+    for (const auto &prec : test.GetMemoryPrecs()) {
+      prec(*mem_hand);
+    }
+
+    test_runner::ExecuteLiftedFunction<State, uint64_t>(
+        new_func, test.target_bytes.length(), &st, mem_hand.get(),
+        [](State *st) { return st->pc.qword; });
```

Comment:
I would like to make this more generic for future arches but this is one of the places where we assume the program counter is called pc which isn't always the case

Reply:
Can always be wrapped in an interface.